### PR TITLE
CI: unpin pymysql, sync min version, and skip failing test

### LINKED ISSUE #36465: CI: sql tests failures casued by pymysql
While working on #36443, I found several sql tests failures related to the latest pymysql and sqlalchemy (could be one of them or both, not sure). See https://travis-ci.org/github/pandas-dev/pandas/jobs/728346186
```
FAILED pandas/tests/io/test_sql.py::TestMySQLAlchemy::test_to_sql_with_negative_npinf[load_iris_data0-input0]

FAILED pandas/tests/io/test_sql.py::TestMySQLAlchemy::test_to_sql_with_negative_npinf[load_iris_data0-input1]

FAILED pandas/tests/io/test_sql.py::TestMySQLAlchemy::test_to_sql_with_negative_npinf[load_iris_data0-input2]

FAILED pandas/tests/io/test_sql.py::TestMySQLAlchemyConn::test_to_sql_with_negative_npinf[load_iris_data0-input0]

FAILED pandas/tests/io/test_sql.py::TestMySQLAlchemyConn::test_to_sql_with_negative_npinf[load_iris_data0-input1]

FAILED pandas/tests/io/test_sql.py::TestMySQLAlchemyConn::test_to_sql_with_negative_npinf[load_iris_data0-input2]

= 6 failed, 76075 passed, 1025 skipped, 1209 xfailed, 6 xpassed, 264 warnings in 1626.88s 
```

```
pymysql                   0.10.1             pyh9f0ad1d_0    conda-forge
sqlalchemy                1.3.19           py37h8f50634_0    conda-forge
```

Now those two packages have been pinned to their minimal support versions.
```
pymysql                   0.7.11                     py_0    conda-forge
sqlalchemy                1.3.0            py37h14c3975_0    conda-forge
```

### Update
The problem is from pymysql. #34431 handled inf dtype for to_sql. But it failed to capture that error while using the latest pymysql as sql driver. The test failed for pymysql>=0.10

### PR DESCRIPTION
**Pull request by fangchenli**
xref #36465

### CODE REVIEW

**Comment by jreback** [comment]
thanks @fangchenli

**Comment by jreback** [comment]
hmm

> FAILED pandas/tests/io/test_sql.py::TestMySQLAlchemy::test_to_sql_with_negative_npinf[load_iris_data0-input2]
FAILED pandas/tests/io/test_sql.py::TestMySQLAlchemyConn::test_to_sql_with_negative_npinf[load_iris_data0-input2]

on the database build

**Comment by fangchenli** [comment]
When one column is `infe0`, df that contains `np.nan` actually could be converted to SQL table. That's why we don't catch any error.

```python
import pandas
import numpy

from sqlalchemy import create_engine

engine = create_engine('mysql+pymysql://root:@localhost/foobar')

table = "foobar"
pandas.DataFrame({'foo': [-numpy.inf, 3], 'infe0': ['bar', 3]}).to_sql(table, engine, index=False)
import pandas.io.sql as sql
res = sql.read_sql_table(table, engine)
print(res)
```

```bash
   foo infe0
0  NaN   bar
1  3.0     3
```

But that column name must be `infe0`. I'm not sure why it's special.

File: pandas/tests/io/test_sql.py [hunk 1]
```diff
@@ -2013,6 +2013,15 @@ def test_to_sql_with_negative_npinf(self, input):
         df = DataFrame(input)
 
         if self.flavor == "mysql":
+            # GH 36465
+            # The input {"foo": [-np.inf], "infe0": ["bar"]} does not raise any error
+            # for pymysql version >= 0.10
+            # TODO: remove this version check after GH 36465 is fixed
+            import pymysql
```
Comment (jreback) [inline pandas/tests/io/test_sql.py:2020]: ok normally would want to do this with an xfail, but ok

File: pandas/tests/io/test_sql.py [hunk 2]
```diff
@@ -2013,6 +2013,15 @@ def test_to_sql_with_negative_npinf(self, input):
...
+
+            if pymysql.VERSION[0:3] >= (0, 10, 0) and "infe0" in df.columns:
+                return
```
Comment (jreback) [inline pandas/tests/io/test_sql.py:2023]: actually can you make this xfail like this: https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexing/test_coercion.py#L116

**Comment by jreback** [comment]
thanks @fangchenli

**Pull request merged**